### PR TITLE
Mobilenet-v2 pot quantization

### PR DESCRIPTION
**Pull request by MilanaShhanukova**
Mobilenet-v2 pot quantization using imagenet-tiny.

### CODE REVIEW

**Comment by ryanloney** [comment]
@MilanaShhanukova please review the contribution guide to align with the structure and format of other notebooks: https://github.com/openvinotoolkit/openvino_notebooks/blob/main/CONTRIBUTING.md

We do not pip install inside the notebooks 

Need to us `!mo` and not `!python /usr/local/lib/python3.7/dist-packages/mo.py`

File: notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "D-frbVLKrkmv"
+      },
+      "source": [
+        "## Prepare the Model\n",
+        "Model preparation stage have the following steps:\n",
```
Comment (MaximProshin) [inline notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb:92]: have => has

File: notebooks/113-image-classification-quantization/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+API](https://docs.openvinotoolkit.org/latest/pot_compression_api_README.html)
+(part of [OpenVINO](https://docs.openvinotoolkit.org/)). We will use [mobilenet-v2](https://pytorch.org/vision/stable/_modules/torchvision/models/mobilenetv2.html) and ImageNet-tiny dataset.
+The code of the tutorial is designed to be extendable to custom models and
+datasets. It consists of the following steps:
+
+- Download and prepare the MRPC model and dataset
+- Define data loading and accuracy validation functionality
+- Prepare the model for quantization
+- Run optimization pipeline
+- Compare performance of the original and quantized models
```
Comment (MaximProshin) [inline notebooks/113-image-classification-quantization/README.md:15]: accuracy comparison step is missed

File: notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+        "                             'type': 'accuracy'}}"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "CclWk-fVd9Wi"
+      },
+      "source": [
+        "## Run Quantization Pipeline\n",
```
Comment (MaximProshin) [inline notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb:314]: I propose to add "and compare the accuracy of the original and quantized models"

File: notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb [hunk 4]
```diff
@@ -0,0 +1,460 @@
...
+      },
+      "source": [
+        "# Quantization of Image Classification Models\n",
+        "\n",
+        "This tutorial demostrates how to apply INT8 quantization to Image Classification model using [Post-training Optimization Tool API](../../compression/api/README.md). The Mobilenet V2 model trained on the ImageNet dataset from Torchvision is used as an example. The code of this tutorial is designed to be extandable to custom model and dataset. It consists of the following steps:\n",
+        "- Install OpenVINO and required tools and packages using PIP manager\n",
+        "- Prepare the model for quantization\n",
+        "- Prepare the ImageNet dataset\n",
+        "- Define data loading and accuracy validation functionality\n",
+        "- Run optimization pipeline\n",
```
Comment (MaximProshin) [inline notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb:16]: add "and compare the accuracy"

**Review by MaximProshin — APPROVED** [review]

**Review by AlexKoff88 — APPROVED** [review]

**Comment by MaximProshin** [comment]
@helena-intel , I think it can be merged if you don't have any additional comments. Can you do that?

**Comment by helena-intel** [comment]
image
@MilanaShhanukova the accuracy of the model on this dataset is extremely low. Can you look into that?

**Comment by MaximProshin** [comment]
I agree. We need to investigate what happened with the accuracy.

**Comment by MilanaShhanukova** [comment]
@helena-intel @ryanloney @MaximProshin As it's mentioned in the notebook in the dataset description, we don't use Imagenet on which the model was trained because of its size. Therefore we use a small tiny-imagenet, so the model gives low results, but the difference between quantized and float is low, so it shows the right quantization process. 

Quote from the notebook: 
"Imagenet tiny with total 200 classes will be used. Note that mobilenet was trained on imagenet with 1000 classes and different preprocessing was used, so the results are expected to be lower for float model." 

Of course, ImageNet can be used, but now it can produce not only memory storage problem due to its size, but also permission problems.

**Comment by helena-intel** [comment]
How does the evaluation work? ImageNet has 1000 classes, Tiny ImageNet 200. I don't see a mapping between ImageNet classes and Tiny ImageNet classes in the notebook.

**Comment by helena-intel** [comment]
Thanks for the updates @MilanaShhanukova . I noticed that the FP accuracy is not the same as is mentioned here https://github.com/chenyaofo/pytorch-cifar-models Can you check that? Maybe there's a difference in preprocessing/transforms.
Also please update the text at the top of the notebook (it still says Tiny ImageNet) including credit for the CIFAR models repo, and check the [contributing guide/coding guidelines](https://github.com/openvinotoolkit/openvino_notebooks/blob/main/CONTRIBUTING.md#coding-guidelines). For consistency, we like  imports at the top, f-strings, no global variables, etc.

**Comment by ryanloney** [comment]
Thank you for the updates @MilanaShhanukova. In addition to Helena's requests above, can you show inference on an image as a step inside the notebook? You can see examples of this in various other notebooks like 001 and 301.

**Comment by MilanaShhanukova** [comment]
@ryanloney  @helena-intel can you please review the notebook again, is there anything to be changed?

**Comment by ryanloney** [comment]
Hi @MilanaShhanukova I will let @paularamo do the final review and decide when it is ready to merge. No further comments from me.

**Review by ryanloney — APPROVED** [review]

**Comment by MaximProshin** [comment]
@paularamo please provide your review. The PR is open for quite some time and we'd like to merge in 22.1 time frame.

**Comment by helena-intel** [comment]
@MilanaShhanukova I know it is not ideal to have to wait so long for a PR to be merged. Thank you for your patience! While waiting for Paula's review, can you check the coding guidelines in https://github.com/openvinotoolkit/openvino_notebooks/blob/main/CONTRIBUTING.md for some final touches?

File: notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb [hunk 5]
```diff
@@ -0,0 +1,446 @@
...
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "rQc-wXjqrEuR"
+      },
+      "source": [
+        "# Quantization of Image Classification Models\n",
+        "\n",
+        "This tutorial demostrates how to apply INT8 quantization to Image Classification model using [Post-training Optimization Tool API](../../compression/api/README.md). The Mobilenet V2 model trained on Cifar10 dataset is used as an example. The code of this tutorial is designed to be extandable to custom model and dataset. It consists of the following steps:\n",
+        "- Install OpenVINO and required tools and packages using PIP manager\n",
```
Comment (paularamo) [inline notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb:12]: The OpenVINO installation was previously performed before opening this notebook. Therefore, I suggest removing this line and mentioning that the libraries are imported and the tools used.

**Comment by paularamo** [comment]
Any way to avoid this kind of outputs? 
image

File: notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb [hunk 6]
```diff
@@ -0,0 +1,446 @@
...
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "DErQofk8tO6c"
+      },
+      "outputs": [],
+      "source": [
+        "class CifarDataLoader(DataLoader):\n",
+        "\n",
```
Comment (paularamo) [inline notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb:130]: I suggest using the docstring format. It provides convenient associate documentation for modules, functions, classes, and methods. Also, I recommend including type hints on it. It is super important that the developer (reader of the notebook) understand each step. See the attached image with docstring and type hints examples:

image

Please check all functions and classes in this notebook.

File: notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb [hunk 7]
```diff
@@ -0,0 +1,446 @@
...
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "CclWk-fVd9Wi"
+      },
+      "source": [
+        "## Run Quantization Pipeline and compare the accuracy of the original and quantized models\n",
+        "Here we define a configuration for our quantization pipeline and run it. \n",
+        "\n",
+        "**Note**: we use built-in `IEEngine` implementation of the `Engine` interface from the POT API for model inference. `IEEngine` is built on top of OpenVINO Python* API for inference and provides basic functionality for inference of simple models. If you have a more complicated inference flow for your model/models you should create your own implementation of `Engine` interface, for example by inheriting from `IEEngine` and extending it."
```
Comment (paularamo) [inline notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb:234]: This is a simple detail, but we would love to see constancy in the word format. May you please define the format for words, such as "Note". This one has different formats throughout the notebook. Thanks.

**Review by paularamo — CHANGES_REQUESTED** [review]
I have added my comments. Please take a look of those. I hope you could find those useful.

File: notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb [hunk 8]
```diff
@@ -0,0 +1,446 @@
...
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "picture_index = 60\n",
+        "\n",
+        "label = all_labels[picture_index]\n",
+        "image = all_batches[picture_index].squeeze().numpy()\n",
+        "image = np.rollaxis(image, 0, 3)\n",
+        "image.shape\n",
```
[the remark this answers is not in the document]
Comment (MilanaShhanukova) [reply notebooks/113-image-classification-quantization/113-image-classification-quantization.ipynb:380]: The picture is displayed just because the inference of it is done later. It was suggested before. To make an inference of 4 pictures will take more code which is not as flexible as inference of one picture chosen by the user. So I suggest doing inference only on one picture. Displaying pictures is not the core idea, in my opinion.

**Review by paularamo — APPROVED** [review]
Thanks for accepting the suggestions.

**Comment by adrianboguszewski** [comment]
@MilanaShhanukova please add a link to the notebook to the main README file.

**Comment by MaximProshin** [comment]
@adrianboguszewski , can you please clarify which link you mean? As far as I can see, the notebooks/README.md file has already been updated in this PR.

**Comment by adrianboguszewski** [comment]
There is one more README.md file in the root directory.

**Comment by MaximProshin** [comment]
@adrianboguszewski , I see. Is it ok now?

**Review by adrianboguszewski — APPROVED** [review]
